### ISSUE TITLE
Warn on duplicate explicit enum values

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn-analyzers#1646]
If you have an enum with explicit values, we should warn if two of those values are identical. These can often be introduced when you're merging in work done in different branches, and can cause failures that seem impossible and are not readily apparent.

**Comment by sharwell** [comment]
:memo: The resolution here is to reference the earlier member. Here is a code sample showing several edge cases along with where the warning should be reported:

```csharp
enum SomeEnum {
  Value1 = 1,
  Value2,
  Value3 = 2, // Warning
  Value4 = Value1, // Allowed
  Value5 = 0,
  Value6, // Warning
  Value7, // Warning
  Value8 = Value6, // Allowed
  Value9 = Value1 | Value1, // Warning
}
```

**Comment by wachulski** [comment]
Looks interesting, consider me working on it.

**Comment by Evangelink** [comment]
What about valid cases that are probably never gonna happen (at least I hope no-one does that):

```csharp
enum MyEnum
{
    Value1 = 1,
	Value2 = 2 - 1, // Warning
	Value3 = Value1 * Value2, // Warning
    Value4 = 42,
	Value5 = Value4 * Value4, // Allowed
}
```
